src/main/java/aquarium/shop/order/OrderServiceImpl.java

```diff
@@ -15,12 +21,25 @@ public class OrderServiceImpl implements OrderService  {
 
     @Override
     public void addOrder(OrderDetails orderDetails) {
+        List<PurchasedProduct> purchasedProductList = new ArrayList<>();
+        int wholePrice = 0;
+        for (HashMap.Entry<Product, Integer> entry : basket.showBasket().getContent().entrySet()) {
+            Product product = entry.getKey();
+            int quantity = entry.getValue();
+            PurchasedProduct purchasedProduct = new PurchasedProduct(product, quantity);
+            purchasedProductList.add(purchasedProduct);
+            wholePrice += (product.getPrice() * quantity);
```

Comment:
Maybe it would be better to use here already calculated **wholePrice** from basket to avoid not needed calculation.